modules/openapi-generator/src/main/resources/swift5/modelInlineEnumDeclaration.mustache

```diff
@@ -1,4 +1,4 @@
-    {{#nonPublicApi}}internal{{/nonPublicApi}}{{^nonPublicApi}}public{{/nonPublicApi}} enum {{enumName}}: {{^isContainer}}{{dataType}}{{/isContainer}}{{#isContainer}}String{{/isContainer}}, {{#useVapor}}Content, Hashable{{/useVapor}}{{^useVapor}}Codable{{/useVapor}}, CaseIterable{{#enumUnknownDefaultCase}}{{#isInteger}}, CaseIterableDefaultsLast{{/isInteger}}{{#isFloat}}, CaseIterableDefaultsLast{{/isFloat}}{{#isDouble}}, CaseIterableDefaultsLast{{/isDouble}}{{#isString}}, CaseIterableDefaultsLast{{/isString}}{{#isContainer}}, CaseIterableDefaultsLast{{/isContainer}}{{/enumUnknownDefaultCase}} {
+    {{#nonPublicApi}}internal{{/nonPublicApi}}{{^nonPublicApi}}public{{/nonPublicApi}} enum {{enumName}}: {{^isContainer}}{{dataType}}{{/isContainer}}{{#isContainer}}String{{/isContainer}}, {{#useVapor}}Content, Hashable{{/useVapor}}{{^useVapor}}Codable{{^isContainer}}{{^isString}}{{^isInteger}}{{^isFloat}}{{^isDouble}}, JSONEncodable{{/isDouble}}{{/isFloat}}{{/isInteger}}{{/isString}}{{/isContainer}}{{/useVapor}}, CaseIterable{{#enumUnknownDefaultCase}}{{#isInteger}}, CaseIterableDefaultsLast{{/isInteger}}{{#isFloat}}, CaseIterableDefaultsLast{{/isFloat}}{{#isDouble}}, CaseIterableDefaultsLast{{/isDouble}}{{#isString}}, CaseIterableDefaultsLast{{/isString}}{{#isContainer}}, CaseIterableDefaultsLast{{/isContainer}}{{/enumUnknownDefaultCase}} {
```

Comment:
@0x0c is it necessary to check is it's not container?
This implementation is different from `modelEnum.mustache`, is it normal?

Reply:
Maybe it's related to https://github.com/OpenAPITools/openapi-generator/pull/11202#discussion_r779967119?
If it's correct being different and you could give a bit of more context, I would appreciate it.

Reply:
If `modelEnumDeclaration.mustache` is container, the enum confirms String. Therefore, we have to check whether the enum is container or not.

Reply:
And this doesn't apply also to `modelInlineEnumDeclaration.mustache`?

Reply:
Oh, sorry I missed type. In previous my post "If `modelEnumDeclaration.mustache` is container ..." is not correct.. "If `modelInlineEnumDeclaration.mustache` is container ..." is correct.

Reply:
This PR looks good to me.
If you could explain why we need to check for container in one but not the other I would appreciate it, because this was something that was always a bit confusing to me

Reply:
> If you could explain why we need to check for container in one but not the other I would appreciate it, because this was something that was always a bit confusing to me

Sorry, I can't answer your question because a code for checking container is already implemented before I modify `modelEnumDeclaration.mustache`. I respect original mustache and don't change the mustache is better I think.

Originally, according to first commit of these files, there is difference between `modelEnum.mustache` and  `modelInlineEnumDeclaration.mustache`.
See [here](https://github.com/OpenAPITools/openapi-generator/blame/c5fefa938a7c3960d7020855b74cdf75ddc31ac7/modules/openapi-generator/src/main/resources/swift5/modelInlineEnumDeclaration.mustache) and [here](https://github.com/OpenAPITools/openapi-generator/blame/c5fefa938a7c3960d7020855b74cdf75ddc31ac7/modules/openapi-generator/src/main/resources/swift5/modelEnum.mustache).

Reply:
I see, thanks for the explanation 👍